client/state/account-recovery/settings/reducer.js

```diff
@@ -77,7 +77,7 @@ const phone = createReducer( null, {
 		convertPhoneResponse( settings.phone ),
 
 	[ ACCOUNT_RECOVERY_SETTINGS_UPDATE_SUCCESS ]: ( state, { target, value } ) =>
-		'phone' === target ? convertPhoneResponse( value ) : state,
+		'phone' === target ? value : state,
```

Comment:
It is intentional. At a latter stage, I realized that the recovery option updating endpoints only returns the status value, but not the updated value. Thus, I had adjusted the corresponding actions in 52e5cdff17daa7a1aa1ca8522212a79afff51a69 accordingly. With it, the value encapsulated by `ACCOUNT_RECOVERY_SETTINGS_UPDATE_SUCCESS` action is no longer in the form of php snake-case name, but the same form as our state tree. Thus calling `convertPhoneResponse` is no longer need in here, but it is still there for `ACCOUNT_RECOVERY_SETTINGS_FETCH_SUCCESS`.

In fact, I have also considered another option: patching the endpoints so that they return the updated value. The patch is D3700. I've struggled which way to pick for a while, and picked the former simply because it requires no update for the endpoints. However, it does sacrifice some data consistency in my opinion, just like you spotted here.

What do you think? Should we reconsider the latter approach?